### ISSUE TITLE
Downloader should set mode back to previous if sdk download is cancelled

### ISSUE THREAD

**Issue by sarahcodes100** [DeerajTheepshi/processing-android#18]
If I:
- open Processing in the Java mode
- change the toggle to Android and get prompted to download the SDK
- start downloading the SDK and cancel it

After we show the dialog that the Android mode has been disabled, we should set the mode to the previously selected one instead of keeping it on Android

**Comment by DeerajTheepshi** [comment]
When we trigger modeChange() : 
- If there's a window that's already open, it's closed and new window is opened.
- If no window is open, a new window is opened. 

While cancelling SDK installation , no window exists, as a result of which 2 windows one android and the other , the previous mode, will be opened. Also , the window is opened by means of super classes which trace back to Processing-Java. 

So I would like to solve this by creating a thread, that listens for the Android window to be opened, and hence change the mode. 

What are your views on this approach @kfrajer @sarahlensing @codeanticode ?

**Comment by sarahcodes100** [comment]
Is it possible to keep a variable for the previous mode when the SDK installation starts and switch the PDE to this mode if the SDK installation cancellation callback happens?

**Comment by DeerajTheepshi** [comment]
Yes. It's possible. Processing base allows various mode selectors :
- defaultMode() - Java
- getModesList().get(index) - Select one from multiple modes

So we can pick one of these, without altering the processing base. Any other option I think, would need a change in processing-base as mode shifting is a property held by processing-base . What are your views @kfrajer @sarahlensing
